File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/shared/message/service/impl/MessageServiceImpl.java
package org.bootstrapbugz.api.shared.message.service.impl;

import org.bootstrapbugz.api.shared.message.service.MessageService;
import org.springframework.context.MessageSource;
import org.springframework.context.i18n.LocaleContextHolder;
import org.springframework.stereotype.Service;

@Service
public class MessageServiceImpl implements MessageService {
  private final MessageSource messageSource;

  public MessageServiceImpl(MessageSource messageSource) {
    this.messageSource = messageSource;
  }

  @Override
  public String getMessage(String code) {
    return messageSource.getMessage(code, null, LocaleContextHolder.getLocale());
  }
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/shared/message/service/impl/MessageServiceImpl.kt
package org.bootstrapbugz.api.shared.message.service.impl

import org.bootstrapbugz.api.shared.message.service.MessageService
import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder
import org.springframework.stereotype.Service

@Service
class MessageServiceImpl(private val messageSource: MessageSource) : MessageService {
    override fun getMessage(code: String): String {
        return messageSource.getMessage(code, null, LocaleContextHolder.getLocale())
    }
}
